js/blocks/components/test/fetch-input.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom/extend-expect';
+import React from 'react';
+import { fireEvent, render, screen, waitForDomChange } from '@testing-library/react';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import FetchInput from '../fetch-input';
+
+jest.mock( '@wordpress/api-fetch' );
+
+const label = 'text-label';
+const value = 'this is a value';
+const displayValue = 'example-display-value';
+const mockOnChange = jest.fn();
+const setup = ( props ) => {
+	const utils = render(
+		<FetchInput { ...props } />
+	);
+	const input = utils.getByLabelText( label );
+	return {
+		input,
+		...utils,
+	};
+};
+const baseProps = {
+	value,
+	displayValue,
+	field: { label },
+	onChange: mockOnChange,
+	getValueFromAPI: ( fromAPi ) => fromAPi,
+	getDisplayValueFromAPI: ( fromApi ) => fromApi,
+};
+
+describe( 'FetchInput', () => {
```

Comment:
The unit tests for these components might be the more interesting part of this PR.